src/components/Pages/ProcessPage/ProcessPage.tsx

```diff
@@ -22,72 +102,122 @@ export class ProcessPage extends BasePage {
-                                </div>
-                            </div>
-                            <div className="stage">
-                                <span className="stage__number marker">
-                                    05
-                                </span>
-                                <div className="stage-body">
-                                    <h3 className="stage__title">Продвижение</h3>
-                                    <p className="stage__description">online / offline</p>
-                                    <span className="stage__detail">&gt;&gt;</span>
-                                </div>
-                            </div>
-                            <div className="stage">
-                                <span className="stage__number marker">
-                                    06
-                                </span>
-                                <div className="stage-body">
-                                    <h3 className="stage__title">Тех. поддержка</h3>
-                                    <p className="stage__description">dsgn / dev / PR</p>
-                                    <span className="stage__detail">&gt;&gt;</span>
-                                </div>
-                            </div>
-                        </div>
-                    </div>
+                    </SmartBreakpoint>
                 </div>
             </section>
         );
     }
+
+    protected setContainer = (r: HTMLElement) => this.stagesContainer = r;
+
+    protected getStages(from: number, count: number): JSX.Element [] {
+        return Array.from(this.stageList)
+            .splice(from, count)
+            .map(({title, subTitle}, i) => {
+                return (
+                    <div className="stage" key={i}>
+                        <span className="stage__number marker">
+                            {toFixed(2, i + from + 1)}
+                        </span>
+                        <div className="stage-body">
+                            <h3 className="stage__title">{title}</h3>
+                            <p className="stage__description">{subTitle}</p>
+                            <span className="stage__detail">&gt;&gt;</span>
+                        </div>
+                    </div>
+                );
+            });
+    }
+
+    protected handleHoverControl = (event: MouseEvent | TouchEvent): void => {
+        if (!this.stagesContainer) {
+            return;
+        }
+
+        const clientX = event.hasOwnProperty("touches")
+            ? (event as TouchEvent).touches[0].clientX
+            : (event as MouseEvent).clientX;
+
+        const left = getElementCoords(this.stagesContainer).left;
+        const width = this.stagesContainer.getBoundingClientRect().width;
+        const cursorOffset = clientX - left;
+
+        // is out of range
+        if (cursorOffset < 0 || cursorOffset > width) {
+            if (this.state.currentIndex !== -1) {
```

Comment:
`if(!A) { ... code } else if (b) { code }` instead of `if(a) { if (b) code } code }`